src/scopes/goals/reasons.js

```diff
@@ -1,30 +1,44 @@
 import { Op } from 'sequelize';
 import { filterAssociation } from './utils';
 
+/*
 const reasonFilter = `
 SELECT
   DISTINCT "Goals"."id"
 FROM "ActivityReports" "ActivityReports"
 INNER JOIN "ActivityReportObjectives" "ActivityReportObjectives"
-ON "ActivityReports"."id" = "ActivityReportObjectives"."activityReportId"
+  ON "ActivityReports"."id" = "ActivityReportObjectives"."activityReportId"
 INNER JOIN "Objectives" "Objective"
-ON "ActivityReportObjectives"."objectiveId" = "Objective"."id"
+  ON "ActivityReportObjectives"."objectiveId" = "Objective"."id"
 INNER JOIN "Goals" "Goals"
-ON "Objective"."goalId" = "Goals"."id"
+  ON "Objective"."goalId" = "Goals"."id"
 WHERE ARRAY_TO_STRING("ActivityReports"."reason", ',')`;
+*/
 
-export function withReasons(reasons) {
+const reasonFilter = (options) => {
+  const useRecipient = options && options.recipientId;
+  return `
+          SELECT DISTINCT g.id
+          FROM "ActivityReports" ar
+          INNER JOIN "ActivityReportGoals" arg ON ar.id = arg."activityReportId"
+          INNER JOIN "Goals" g ON arg."goalId" = g.id
+          INNER JOIN "Grants" gr ON g."grantId" = gr."id"
+          WHERE ${useRecipient ? `gr."recipientId' = ${options.recipientId} AND ` : ''}
```

Comment:
We will want to escape the `options.recipientId` before including it in SQL (unless it happens further down somewhere!)

Reply:
Escape ' ' added.